src/cli/index.js

```diff
@@ -93,38 +93,57 @@ export function main({
   commander.option('--non-interactive', 'do not show interactive prompts');
   commander.option('--scripts-prepend-node-path [bool]', 'prepend the node executable dir to the PATH in scripts');
 
-  // get command name
-  let commandName: string = args.shift() || 'install';
-
   // if -v is the first command, then always exit after returning the version
-  if (commandName === '-v') {
+  if (args[0] === '-v') {
     console.log(version.trim());
     process.exitCode = 0;
     return;
   }
 
-  if (commandName === '--help' || commandName === '-h') {
-    commandName = 'help';
+  // get command name
+  const firstNonFlagIndex = args.findIndex(arg => !arg.startsWith('-'));
```

Comment:
Are there unit tests covering all this stuff?

Reply:
Yup :)